Ruby/Ruby.sublime-syntax

```diff
@@ -479,11 +533,30 @@ contexts:
       scope: variable.other.constant.ruby
 
   well-known-methods:
-    - match: '\b(initialize|new|loop|include|extend|prepend|raise|fail|attr_reader|attr_writer|attr_accessor|attr|catch|throw|module_function|public|protected|private)\b(?![?!])'
+    # exceptions
+    - match: \bcatch\b(?![?!])
+      scope: keyword.control.exception.catch.ruby
+      push: function-call-arguments
+    # flow
+    - match: \braise\b(?![?!])
+      scope: keyword.control.flow.raise.ruby
```

Comment:
`raise` and `throw` should be unified to a single name because they behave in the same or at a least similar way. In #1228 I proposed `keyword.control.throw` because of the number of C++-like languages.
I see that Ruby has a `throw` keywor as well. How do they relate?

(Unless you're saying we should always use the keyword's name at the fourth subscope position, which I'd like to discuss first.)

Reply:
Both keywords are explicitly added to the scope names in the following languages for instance.

`keyword.control.throw` is not used anywhere. At least `throw` should be part of the `keyword.control.flow` scope.

Java

https://github.com/sublimehq/Packages/blob/f60f80afdb66e317c3f2a0852861171ab677621f/Java/Java.sublime-syntax#L629

Python

https://github.com/sublimehq/Packages/blob/f60f80afdb66e317c3f2a0852861171ab677621f/Python/Python.sublime-syntax#L86

Reply:
Ah, yes, that was a typo. I meant to write `keyword.control.flow.throw`.

I'm just worried about us discussing scope names and eventually leading to a conclusion somewhat but then not acting on it when given the chance.

Reply:
We came arround with adding dedicated scopes for `die`, `panic`, `exit`, ... even though they basically have the same meaning in the one way or the other. But with `raise` and `throw` we argue to use the same 4th level scopes which might propably never be addressed by color schemes at this depth.

That's just weird. It just feels more natural to keep the first 3 levels balanced and use the keyword as 4th level for those who really want to use different scopes for each keyword.

I think its worth to keep something like `elif` or `elsif` or `elseif` with the same scope.

Reply:
I don't remember us deciding on unsing different 4th level scopes for `die` or `panic`. My proposal (https://github.com/sublimehq/Packages/issues/1228#issuecomment-461077685) only includes `panic` as well. Maybe you misunderstood something?

Reply:
> ... I don't remember us deciding on using different 4th level scopes ...

Oh. So I missed something. Keywords were a while ago.

> The actual keyword could then be added in the 5th level.

1. The distinction between "meaning" and "keyword" is sometimes hared to see.
2. Not sure if scoping "each keyword" with its dedicated scope is really useful. A reasonable number of groups to target different types of keywords is propably enough.

Reply:
I suggest we discuss that in #1228 separately.